src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public x: number,
+    public y: number,
+    public z: number,
+  ) {
+    const isLessthenOne = x <= 0 || y <= 0 || z <= 0;
+    const isUnvalidX = x >= y + z;
+    const isUnvalidY = y >= x + z;
+    const isUnvalidZ = z >= x + y;
 
-export class Circle {
+    if (isLessthenOne || isUnvalidX || isUnvalidY || isUnvalidZ) {
+      throw new Error('Some of the values is not valid');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.x + this.y + this.z) / 2;
+    const result = Math.sqrt(s * ((s - this.x) * (s - this.y) * (s - this.z)));
+
+    return Math.floor(result * 100) / 100;
```

Comment:
As you use `Math.floor(value * 100) / 100;` expression in every class, why don't you move it to separate function?